### PR TITLE
[2201.3.x] Fix several completions related LS bugs

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/38925
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/38897
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/38896

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java [hunk 1]
```diff
@@ -53,6 +53,14 @@ public RemoteMethodCallActionNodeContext() {
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, RemoteMethodCallActionNode node)
             throws LSCompletionException {
         List<LSCompletionItem> completionItems = new ArrayList<>();
+        if (onSuggestClients(node, context)) {
+            // Covers following:
+            // 1. cl<cursor>->func()
+            completionItems.addAll(this.expressionCompletions(context));
+            this.sort(context, node, completionItems);
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java:60]: Shall we improve the sorting in this context as well
image

**Review by malinthar — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java:60]: Done. Addressed. Also addressed the same for client resource access action

**Review by malinthar — APPROVED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/action_node_context/source/client_resource_access_action_source20.bal [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+public function testClient(string param1, int count) returns error? {
+    MyClient myClient = new ();
+    string response = my->/user/[1]/path1/path2/["path3"];
+}
+
+function getClient() returns MyClient {
+    return new MyClient();
+}
```
Comment (mindula) [inline language-server/modules/langserver-core/src/test/resources/completion/action_node_context/source/client_resource_access_action_source20.bal:45]: Add a new line

**Review by mindula — APPROVED** [review]
LGTM

**Pull request merged**